### PR TITLE
feat(core): emit addedUserIds on invitation accept and experience-flow jit

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary
Phase 3 of [LOG-13464](https://linear.app/silverhand/issue/LOG-13464). Relates to #6757.

### What changed
- `packages/core/src/routes/organization-invitation/index.ts` — invitation accept emits `addedUserIds: [acceptedUserId]` on `Organization.Membership.Updated`. Gated by `EnvSet.values.isDevFeaturesEnabled`. Also adds `buildManagementApiContext(ctx)` to the emit (the route is a Management API call but historically omitted the management-context spread; matches Phase 1's user-route pattern at `routes/organization/user/index.ts:93-98` and is what lets the integration-test harness key webhook deliveries by `method + matchedRoute`).
- `packages/core/src/routes/experience/classes/libraries/provision-library.ts` — the JIT provisioning loop emits `addedUserIds: [payload.userId]` per organization on `Organization.Membership.Updated`. Gated. No `buildManagementApiContext` (experience-flow emission with no meaningful `matchedRoute`).
- `packages/integration-tests/src/tests/api/hook/` — split `hook.trigger.data.test.ts` (was 412 lines, near the 400-line test-file lint cap) into:
  - `hook.trigger.data.test.ts` — user / role / scope describes + a non-org coverage assertion.
  - `hook.trigger.organization.test.ts` — org / org-scope / org-role describes + a new invitation-accept describe block + an org-scoped coverage assertion.
  - `webhook-fixtures.ts` — `createDataHookFixture(port, hookName)` factory owning mock-server + webhook subscription lifecycle, plus a `partitionManagementApiHookKeys()` helper shared by both files' coverage assertions.
  - `test-cases-organization.ts` — org / org-scope / org-role test-case arrays extracted from `test-cases.ts`.

### Out of scope (intentional)
`packages/core/src/routes/interaction/*` is the deprecated interaction API surface. The three legacy emit sites (`submit-interaction.ts:203`, `single-sign-on.ts:397`, `single-sign-on.ts:466`) stay on the `{ organizationId }`-only payload. Consumers on interaction-API flows must migrate to the experience API for the new contract.

### Notes
- No changeset in this PR — Phase 3 is fully behind `EnvSet.values.isDevFeaturesEnabled`. The release task ([LOG-13467](https://linear.app/silverhand/issue/LOG-13467)) removes the guard and lands the consolidated changeset that promotes the new fields to GA.
- Both test files share port 9999 (relies on `jest --runInBand` per `pnpm test:api`); this is documented in `webhook-fixtures.ts`.
- The `// TODO: Remove this check` short-circuit in the org describe block is preserved verbatim from master; Phase 2 ([#8834](https://github.com/logto-io/logto/pull/8834)) removes it as part of moving app-membership cases to dual-mode.

## Testing
Integration tests

## Checklist
- [ ] \`.changeset\`
- [ ] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts [hunk 1]
```diff
@@ -154,6 +155,8 @@ export class ProvisionLibrary {
     for (const { organizationId } of provisionedOrganizations) {
       this.ctx.appendDataHookContext('Organization.Membership.Updated', {
         organizationId,
+        ...(EnvSet.values.isDevFeaturesEnabled &&
+          truncateMembershipDelta({ addedUserIds: [payload.userId] })),
       });
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/routes/experience/classes/libraries/provision-library.ts:160]: Addressed in 882e1602a. `provisionOrganizations` (`packages/core/src/libraries/user.ts`) now snapshots pre-existing memberships via the new `getExistingOrganizationIds(userId, orgIds[])` helper in `user-relations.ts` **before** the insert, then filters the return to only the JIT orgs the user was newly added to. The caller's webhook loop iterates this filtered list, so `addedUserIds` reflects truly-new memberships only.

Gated behind `EnvSet.values.isDevFeaturesEnabled` — when the flag is off, the function returns the full `jitOrganizations` list (master's webhook-for-every-org behavior is preserved). The gate flips on at GA via [LOG-13467](https://linear.app/silverhand/issue/LOG-13467).

File: packages/core/src/routes/organization-invitation/index.ts [hunk 2]
```diff
@@ -162,7 +164,10 @@ export default function organizationInvitationRoutes<T extends ManagementApiRout
 
       const { organizationId } = result;
       ctx.appendDataHookContext('Organization.Membership.Updated', {
+        ...buildManagementApiContext(ctx),
         organizationId,
+        ...(EnvSet.values.isDevFeaturesEnabled &&
+          truncateMembershipDelta({ addedUserIds: [acceptedUserId] })),
       });
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/routes/organization-invitation/index.ts:198]: Addressed in 882e1602a. The invitation accept route now pre-checks membership via `getExistingUserIds(organizationId, [acceptedUserId])` **before** delegating to `updateStatus` (which inserts via ON CONFLICT DO NOTHING, after which the pre-state is lost).

When the dev-features gate is on and the user was already a member, the `Organization.Membership.Updated` webhook is **skipped entirely** rather than emitted with an empty delta — the event name promises a membership change, and accepting an invitation that produces no row insert isn't one. Master behavior (always emit) is preserved when the gate is off.

New integration test covers both branches: `hook.trigger.organization.test.ts` → `does not emit ... when the accepted user is already a member (dev-features on)`.

File: packages/integration-tests/src/tests/api/hook/hook.trigger.organization.test.ts [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    const user = await userApi.create({ primaryEmail: invitation.invitee });
+
+    await invitationApi.updateStatus(invitation.id, OrganizationInvitationStatus.Accepted, user.id);
+
+    const hook = await getWebhookResult('PUT /organization-invitations/:id/status');
+    expect(hook?.payload.event).toBe('Organization.Membership.Updated');
+    expect(hook?.payload).toMatchObject({
+      organizationId: organization.id,
+      ...(isDevFeaturesEnabled && { addedUserIds: [user.id] }),
+    });
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/integration-tests/src/tests/api/hook/hook.trigger.organization.test.ts]: Addressed in 882e1602a. The happy-path test now branches on `isDevFeaturesEnabled`:
- **Dev features on**: asserts `toMatchObject({ organizationId, addedUserIds: [user.id] })`.
- **Dev features off**: asserts `toMatchObject({ organizationId })` AND explicitly `expect(hook?.payload).not.toHaveProperty('addedUserIds')` — regression-protecting the gate.

Both branches also assert `not.toHaveProperty('removedUserIds')` (only one user added; omit-empty contract). Added a sibling test for the already-a-member edge case that follows the same dual-mode pattern (no webhook fires when gate on; legacy emit when gate off).

**Review by wangsijie — APPROVED** [review]
